powershell-module/Az.Tools.Migration/Classes/Classes.ps1

```diff
@@ -160,7 +157,7 @@ class UpgradeResult
 
         if ($Plan.PlanSeverity -eq [DiagnosticSeverity]::Warning)
         {
-            $this.UpgradeResult = [UpgradeResultReasonCode]::UpgradedWithWarnings
+            $this.UpgradeResult = [UpgradeResultReasonCode]::UnableToUpgrade
```

Comment:
Does it mean the upgrade check is stricter than previous?

Reply:
Yes, this was intentional. The previous warning could still upgrade the cmdlet (upgrade with warning), this new reason code cannot upgrade a dynamic parameter. At the moment we now only have 1 warning condition since splatted parameters (the previous warning) are now supported. So if we add new warnings, this logic may need to be changed to be more flexible.